src/main/java/de/medizininformatikinitiative/flare/model/fhir/Resource.java

```diff
@@ -3,9 +3,13 @@
 import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
 
 @JsonIgnoreProperties(ignoreUnknown = true)
-public record Resource(String id, Reference patient, Reference subject) {
+public record Resource(String id, Reference patient, Reference subject, String resourceType) {
 
     public String patientId() {
         return patient != null ? patient.id() : subject != null ? subject.id() : id;
```

Comment:
Because `patient.id()` returns an `Optional<String>` now, we will return `Optional<String>` from `patientId()` also. Please note that the resource `id` can also be `null`.